tests/test_single_topology.py

```diff
@@ -1708,14 +1708,15 @@ def test_hif2a_end_state_symmetry_nightly_test():
             amm_rev = AtomMapMixin(mol_b, mol_a, core[:, ::-1])
             fused_map = np.concatenate(
                 [
-                    np.array([[x, y] for x, y in zip(amm_fwd.a_to_c, amm_rev.b_to_c)]),
+                    np.array([[x, y] for x, y in zip(amm_fwd.a_to_c, amm_rev.b_to_c)], dtype=np.int32).reshape(-1, 2),
                     np.array(
                         [
                             [x, y]
                             for x, y in zip(amm_fwd.b_to_c, amm_rev.a_to_c)
                             if x not in core[:, 0] and y not in core[:, 1]
-                        ]
-                    ),
+                        ],
+                        dtype=np.int32,
+                    ).reshape(-1, 2),
```

Comment:
nit/non-blocking: I think this could be simplified by moving the array construction to the outside:

```python
            core_map = [(x, y) for x, y in zip(amm_fwd.a_to_c, amm_rev.b_to_c)]
            dummy_map = [
                (x, y) for x, y in zip(amm_fwd.b_to_c, amm_rev.a_to_c) if x not in core[:, 0] and y not in core[:, 1]
            ]
            fused_map = np.array(core_map + dummy_map).reshape(-1, 2)
```